dissect/target/plugins/browsers/chrome.py

```diff
@@ -1,65 +1,31 @@
-from dissect.sql import sqlite3
-from dissect.sql.exceptions import Error as SQLError
-from dissect.util.ts import webkittimestamp
-
-from dissect.target.exceptions import FileNotFoundError, UnsupportedPluginError
-from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
-from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target import Target
 from dissect.target.plugin import Plugin, export
-from dissect.target.plugins.browsers.browser import (
-    GENERIC_HISTORY_RECORD_FIELDS,
-    try_idna,
-)
-
-ChromeBrowserHistoryRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
-    "browser/chrome/history", GENERIC_HISTORY_RECORD_FIELDS
+from dissect.target.plugins.browsers.chromium import (
+    ChromiumPlugin,
+    ChromiumBrowserHistoryRecord,
 )
 
 
 class ChromePlugin(Plugin):
-    """Chrome browser plugin."""
+    """Chrome browser plugin.
 
-    __namespace__ = "chrome"
+    ChromiumPlugin contains all logic and configuration.
+    """
 
-    DIRS = [
-        "AppData/Local/Google/Chrome/User Data/Default",
-        "AppData/Local/Google/Chrome/continuousUpdates/User Data/Default",
-        "Local Settings/Application Data/Google/Chrome/User Data/Default",
-        "AppData/local/Google/Chromium/User Data/Default",
-        ".config/google-chrome/Default",
-        "snap/chromium/common/chromium/Default",
-    ]
+    __namespace__ = "chrome"
 
-    def __init__(self, target):
+    def __init__(self, target: Target):
         super().__init__(target)
-
-        self.users_dirs = []
-        for user_details in self.target.user_details.all_with_home():
-            for d in self.DIRS:
-                cur_dir = user_details.home_path.joinpath(d)
-                if not cur_dir.exists():
-                    continue
-                self.users_dirs.append((user_details.user, cur_dir))
+        self.chromium_hist: ChromiumPlugin = ChromiumPlugin(self.target)
```

Comment:
Can you not subclass `ChromePlugin` from `ChromiumPlugin` and have that work?

I'm not sure if this works, but something like:

```
ChromeBrowserHistoryRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
    "browser/chrome/history", GENERIC_HISTORY_RECORD_FIELDS
)

class ChromePlugin(ChromiumPlugin):
    __namespace__ = "chrome"

    DIRS = []  # Setting this as a class attribute would allow the Chromium plugin to just do self.DIRS and have that work for each subclass
    HISTORY_RECORD = ChromeBrowserHistoryRecord  # Setting this as a class attribute would allow us to have individually named browser records ("browser/chrome/history") but allow the ChromiumPlugin to do self.HISTORY_RECORD(...)

    # Re-export on the ChromePlugin so that it has the correct record associated with it, but simply call the history function of the super class (ChromiumPlugin)
    @export(record=ChromeBrowserHistoryRecord)
    def history(self):
        yield from super().history()
```

Reply:
This would also enable separating Chrome from Chromium itself, since the "default" ChromiumPlugin could have a `DIRS` and `HISTORY_RECORD` that point to the actual Chromium directories, instead of those directories being part of the ChromePlugin